src/components/App/App.js

```diff
@@ -2,25 +2,59 @@ import React, { useState, useEffect } from 'react';
 import { getTips, addTip } from '../../ApiCalls';
 import TipJar from '../TipJar/TipJar';
 import Form from '../Form/Form';
+import Error from '../Error/Error';
+import Loader from '../Loader/Loader';
 import './App.css';
 
 const App = () => {
   const [advice, setAdvice] = useState([]);
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState('Loading tips...');
+
+  // useEffect(() => {
+  //   getTips()
+  //   .then(data => setAdvice(data.rows))
+  // }, [])
 
   useEffect(() => {
-    getTips().then(data => setAdvice(data.rows))
-  }, [])
+    const fetchData = async () => {
+      setError('');
+ 
+      try {
+        const result = await getTips();
+        setAdvice(result.rows);
+      } catch (error) {
+        setError('Oops, problem loading tips. Please refresh the page.');
+      }
+ 
+      setLoading('');
+    };
+ 
+    fetchData();
+  });
 
   const handleAddTip = (newTip) => {
     setAdvice([...advice, newTip])
     addTip(newTip)
   }
 
+  // const validateInputValues = (title, description) => {
+  //   if (!title && !description) {
+  //     setError('Please fill out title & description fields.')
+  //   } else {
+  //     setError('')
+  //   }
+  // } 
+
   return (
     <main className='main'>
       <header><h1>Turing Tip Jar</h1></header>
       <Form handleAddTip={handleAddTip}/>
+      {loading ? <Loader loading={loading}/> : 
```

Comment:
I think we could refactor this into one conditional so we aren't writing rendering the Tip jar in two different components something like `!error && !loading && <TipJar />`

Reply:
You're right about the loading property. That's exactly what Leta discussed in class. I removed that property and any mention of it in the useEffect method. I changed the Loader conditional rendering to check for !advice.length. I also added the [ ] back into our useEffect method.